### PR TITLE
Test data generator

### PR DESCRIPTION
**Pull request by ghost**
DataGenerator class.

See #61 for issue.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/63" target="_blank">ReviewNinja

**Comment by 0xsuu** [comment]
Resolve conflicts please.

**Comment by 0xsuu** [comment]
Considering create something to provide a correct result for the data you generated. quote from @BlueCocoa

**Comment by 0xsuu** [comment]
#include <chrono>

File: llama_cpp/example/sort_test/DataGenerator.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+namespace llama {
+    class DataGenerator {
+    public:
+        DataGenerator() {}
+
+        void generateSequenceVector(std::vector<int> *targetVector, int count=5, bool reverse=true);
+    };
+}
+
+#endif /* DataGenerator_hpp */
```
Comment (0xsuu) [inline llama_cpp/example/sort_test/DataGenerator.h]: This won't pass the style check, pay attention.

Comment (0xsuu) [inline llama_cpp/example/sort_test/DataGenerator.h]: Should always build all before commit/push.

File: llama_cpp/llama_cpp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,2939 @@
+// !$*UTF8*$!
```
Comment (0xsuu) [inline llama_cpp/llama_cpp.xcodeproj/project.pbxproj:1]: Merge with master and remove _all_ the Xcode related files

File: llama_cpp/src/DataGenerator/DataGenerator.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    std::normal_distribution<double> distribution(5.0, 2.0);
+
+    for (int i = 0; i < count; i++) {
+        targetVector->push_back(distribution(generator) > 5.0 ? true:false);
+    }
+}
+
+template<>
+void llama::RandomData::generateRandomData<double>(std::vector<double> *targetVector, int count) {
+    unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
```
Comment (0xsuu) [inline llama_cpp/src/DataGenerator/DataGenerator.cpp]: Replace this

File: llama_cpp/src/DataGenerator/DataGenerator.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+#include <stdio.h>
+#include <vector>
+#include <random>
+
+namespace llama {
+class RandomData {
+ public:
+    RandomData() {}
+
+
```
Comment (0xsuu) [inline llama_cpp/src/DataGenerator/DataGenerator.h:30]: Remove abundant blank line here.

File: llama_cpp/src/DataGenerator/DataGenerator.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    std::default_random_engine generator(seed);
+    std::normal_distribution<double> distribution(10.0, 4.0);
+
+    for (int i = 0; i < count; i++) {
+        targetVector->push_back(distribution(generator));
+    }
+}
+
+template<>
+void llama::RandomData::generateRandomData<bool>(std::vector<bool> *targetVector, int count) {
```
Comment (0xsuu) [inline llama_cpp/src/DataGenerator/DataGenerator.cpp:46]: We need string, not bool, but your solution is correct. :+1:

**Comment by 0xsuu** [comment]
@BlueCocoa Please review this.

**Pull request merged**